keras/applications/imagenet_utils.py

```diff
@@ -18,26 +18,32 @@ def _preprocess_numpy_input(x, data_format, mode):
         x -= 1.
         return x
 
-    if data_format == 'channels_first':
-        if x.ndim == 3:
+    if mode == 'torch':
```

Comment:
I'm not sure there's anything torch-specific about this scheme (unlike the `caffe` scheme which was used by every pretrained model we had that came from Caffe). Or is it really torch-specific?

Reply:
To be precise, this is not torch-specific. This scheme is originated from [Torch ResNets](https://github.com/facebook/fb.resnet.torch/blob/master/pretrained/classify.lua) and many Torch implementations ([ResNeXt](https://github.com/facebookresearch/ResNeXt), [DenseNet](https://github.com/liuzhuang13/DenseNet), ...) follow the scheme.